File: WCPMVCCS/Controllers/HomeController.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Web;
using System.Web.Mvc;

namespace WCPMVCCS.Controllers
{
    public class HomeController : Controller
    {
        public ActionResult Index()
        {

            ViewBag.WCPPDetectionScript = Neodynamic.SDK.Web.WebClientPrint.CreateWcppDetectionScript(Url.Action("ProcessRequest", "WebClientPrintAPI", null, HttpContext.Request.Url.Scheme), HttpContext.Session.SessionID);

            return View();
        }

        public ActionResult Samples()
        {
            return View();
        }

        public ActionResult PrintersInfo()
        {
            ViewBag.WCPScript = Neodynamic.SDK.Web.WebClientPrint.CreateScript(Url.Action("ProcessRequest", "WebClientPrintAPI", null, HttpContext.Request.Url.Scheme), "", HttpContext.Session.SessionID);

            return View();
        }
    }
}

File: WCPMVCVB/Controllers/HomeController.vb
Public Class HomeController
    Inherits System.Web.Mvc.Controller
    Function Index() As ActionResult
        ViewData("WCPPDetectionScript") = Neodynamic.SDK.Web.WebClientPrint.CreateWcppDetectionScript(Url.Action("ProcessRequest", "WebClientPrintAPI", Nothing, HttpContext.Request.Url.Scheme), HttpContext.Session.SessionID)

        Return View()
    End Function

    Function Samples() As ActionResult

        Return View()
    End Function

    Function PrintersInfo() As ActionResult
        ViewData("WCPScript") = Neodynamic.SDK.Web.WebClientPrint.CreateScript(Url.Action("ProcessRequest", "WebClientPrintAPI", Nothing, HttpContext.Request.Url.Scheme), "", HttpContext.Session.SessionID)

        Return View()
    End Function
End Class
